punx/h5tree.py

```diff
@@ -320,10 +319,14 @@ def __render(obj, rank, key, indents):
             elif rank < 4:
                 # this replaces a lot of code: if rank == ...
                 indices = ", ".join([str(key)] + (":" * (rank - 1)).split())
-                part = eval("obj[%s]" % indices)
+                try:
+                    part = eval(f"obj[{indices}]")
+                except OSError as exc:
+                    return f"indices={indices}, obj={obj}, exc={exc}"
+
                 item = self._renderNdArray(part, indents + "  ")  # recursion
             else:
-                item = "rank=%d" % (rank - 1)
+                item = f"rank={rank - 1}"
```

Comment:
```
                item = f"rank={(rank - 1):d}"
```
This is the equivalent format string, but maybe removing coercion to integer is a good idea.

Reply:
Agree.

Reply:
... that is, leaving as `item = f"rank={rank - 1}"` without the coercion